src/daemon/daemon.cpp

```diff
@@ -320,16 +320,17 @@ std::unordered_map<std::string, mp::VMSpecs> load_db(const mp::Path& data_path,
             mounts[json["target_path"].toString().toStdString()] = mp::VMMount{json};
         }
 
-        reconstructed_records[key] = {num_cores,
-                                      mp::MemorySize{mem_size.empty() ? mp::default_memory_size : mem_size},
-                                      mp::MemorySize{disk_space.empty() ? mp::default_disk_size : disk_space},
-                                      default_mac_address,
-                                      MP_JSONUTILS.read_extra_interfaces(record),
-                                      ssh_username,
-                                      static_cast<mp::VirtualMachine::State>(state),
-                                      mounts,
-                                      deleted,
-                                      metadata};
+        reconstructed_records[key] = {
+            num_cores,
+            mp::MemorySize{mem_size.empty() ? mp::default_memory_size : mem_size},
+            mp::MemorySize{disk_space.empty() ? mp::default_disk_size : disk_space},
+            default_mac_address,
+            MP_JSONUTILS.read_extra_interfaces(record).value_or(std::vector<mp::NetworkInterface>{}),
```

Comment:
I think the origninal `MP_JSONUTILS.read_extra_interfaces` function which returns `std::vector<mp::NetworkInterface>` instead of `std::optional<std::vector<mp::NetworkInterface>>` does this already, right? When `record.contains("extra_interfaces")` is false, then the function returns an empty vector, which is essentially the `std::optional` + `value_or(std::vector<mp::NetworkInterface>{})` does.

Reply:
Thanks @georgeliao! The point is that `read_extra_interfaces()` was modified to distinguish both cases (empty vector and absent value). This is needed in the snapshot code. I have chosen to do it this way because I didn't want two functions in `json_utils.cpp`.

Reply:
In the case of snapshot, I think the ternary if can do the job based on the original `read_extra_interfaces` function, that makes `record.contains("extra_interfaces") ? MP_JSONUTILS.read_extra_interfaces(json) : desc.extra_interfaces`. However, there is an additional check of `record.contains("extra_interfaces") `, so in summary, maybe the `std::optional` approach is fine here.